File: Commercial/Commercial_ControllerCS/Commercial_ControllerCS/ElevatorDoor.cs
using System;
using static System.Threading.Thread;
using static System.Console;

namespace Commercial_ControllerCS
{
    class ElevatorDoor
    {
        #region PROPERTIES
        public string Status { get; private set; }
        #endregion

        #region CONSTRUCTOR
        public ElevatorDoor(string status_)
        {
            Status = status_;
        }
        #endregion

        #region METHODS
        public void OpenDoor()
        {
            Status = "opened";
            WriteLine("\nDoor has opened");
            Sleep(Program.SLEEP_TIME);
        }

        public void CloseDoor()
        {
            Status = "closed";
            WriteLine("Door has closed\n");
            Sleep(Program.SLEEP_TIME);
        }
        #endregion
    }
}

File: Commercial/Commercial_ControllerGo/ElevatorDoor.go
package main

import (
	"fmt"
	"time"
)

type ElevatorDoor struct {

	// PROPERTIES
	Status string
}

// METHODS
func (door *ElevatorDoor) OpenDoor() {

	door.Status = "opened"
	fmt.Println("\n\nDoor has opened")
	time.Sleep(time.Second)
}

func (door *ElevatorDoor) CloseDoor() {

	door.Status = "closed"
	fmt.Println("Door has closed")
	time.Sleep(time.Second)
}
